backends/dnf/pk-backend-dnf.c

```diff
@@ -150,12 +151,34 @@ pk_backend_setup_dnf_context (DnfContext *context, GKeyFile *conf, const gchar *
 	dnf_context_set_cache_dir (context, cache_dir);
 	solv_dir = g_build_filename (destdir, "/var/cache/PackageKit", release_ver, "hawkey", NULL);
 	dnf_context_set_solv_dir (context, solv_dir);
-	repo_dir = g_build_filename (destdir, "/etc/yum.repos.d", NULL);
-	dnf_context_set_repo_dir (context, repo_dir);
 	lock_dir = g_build_filename (destdir, "/var/run", NULL);
 	dnf_context_set_lock_dir (context, lock_dir);
 	dnf_context_set_rpm_verbosity (context, "info");
 
+	/* Set repos and vars dirs */
+	repo_dirs = dnf_context_get_repos_dir (context);
+	if (repo_dirs[0] != NULL) {
+		g_auto(GStrv) full_repo_dirs = NULL;
+		guint len = 1;
+		while (repo_dirs[len] != NULL)
+			len++;
+		full_repo_dirs = g_new0(gchar*, len + 1);
+		for (guint diriter = 0; diriter < len; diriter++)
```

Comment:
We usually use `i` as a for loop counter

Reply:
Done.